### ISSUE TITLE
Fails to correctly parse a Wikibooks page

### ISSUE THREAD

**Issue by joepie91** [spencermountain/wtf_wikipedia#38]
- **Library version:** 0.5.1g
- **Node.js version:** v6.4.0

I encountered two problems while attempting to parse the markup on [this page](https://en.wikibooks.org/w/index.php?title=QEMU/Monitor&action=edit):
- The parser stops parsing after the `gdbserver` section.
- Code blocks are not indicated and such, and are just seen as regular text.

The original input:

``` mediawiki
When QEMU is running, it provides a monitor console for interacting with QEMU. Through various commands, the monitor allows you to inspect the running guest OS, change removable media and USB devices, take screenshots and audio grabs, and control various aspects of the virtual machine.

The monitor is accessed from within QEMU by holding down the Ctrl and Alt keys (or whatever the "mouse grab" keystrokes are), and pressing <code>Shift-2</code>. Once in the monitor, <code>Shift-1</code> switches back to the guest OS. Typing <code>help</code> or <code>?</code> in the monitor brings up a list of all commands.
Alternatively the monitor can be redirected to using the <code>-monitor <dev></code> command line option.
Using <code>-monitor stdio</code> will send the monitor to the standard output, this is most useful when using qemu on the command line.

==Help and information==

===help===
* <tt>help [''command'']</tt> or <tt>? [''command'']</tt>
With no arguments, the help command lists all commands available. For more detail about another command, type <tt>help ''command''</tt>, e.g.
 (qemu) help info
On a small screen / VM window, the list of commands will scroll off the screen too quickly to let you read them. To scroll back and forth so that you can read the whole list, hold down the control key and press Page Up and Page Down.

===info===
* <tt>info ''option''</tt>
Show information on some aspect of the guest OS. Available options are:
* <tt>block</tt> &ndash; block devices such as hard drives, floppy drives, cdrom
* <tt>blockstats</tt> &ndash; read and write statistics on block devices
* <tt>capture</tt> &ndash; active capturing (audio grabs)
* <tt>history</tt> &ndash; console command history
* <tt>irq</tt> &ndash; statistics on interrupts (if compiled into QEMU)
* <tt>jit</tt> &ndash; statistics on QEMU's Just In Time compiler
* <tt>kqemu</tt> &ndash; whether the kqemu kernel module is being utilised
* <tt>mem</tt> &ndash; list the active virtual memory mappings
* <tt>mice</tt> &ndash; mouse on the guest that is receiving events
* <tt>network</tt> &ndash; network devices and VLANs
* <tt>pci</tt> &ndash; PCI devices being emulated
* <tt>pcmcia</tt> &ndash; [[w:PC Card|PCMCIA card]] devices
* <tt>pic</tt> &ndash; state of i8259 (PIC)
* <tt>profile</tt> &ndash; info on the internal profiler, if compiled into QEMU
* <tt>registers</tt> &ndash; the CPU registers
* <tt>snapshots</tt> &ndash; list the VM snapshots
* <tt>tlb</tt> &ndash; list the TLB ([[w:Translation lookaside buffer|Translation Lookaside Buffer]]), i.e. mappings between physical memory and virtual memory
* <tt>usb</tt> &ndash; USB devices on the virtual USB hub
* <tt>usbhost</tt> &ndash; USB devices on the host OS
* <tt>uuid</tt> &ndash; Unique id of the VM
* <tt>version</tt> &ndash; QEMU version number
* <tt>vnc</tt> &ndash; [[w:Virtual Network Computing|VNC]] information

==Devices==

===change===
* <tt>change ''device'' ''setting''</tt>
The <code>change</code> command allows you to change removable media (like CD-ROMs), change the display options for a VNC, and change the password used on a VNC.

When you need to change the disc in a CD or DVD drive, or switch between different [[w:ISO image|.iso]] files, find the name of the CD or DVD drive using <code>info</code> and use <code>change</code> to make the change.
 (qemu) info block
 ide0-hd0: type=hd removable=0 file=/path/to/winxp.img
 ide0-hd1: type=hd removable=0 file=/path/to/pagefile.raw
 ide1-hd1: type=hd removable=0 file=/path/to/testing_data.img
 ide1-cd0: type=cdrom removable=1 locked=0 file=/dev/sr0 ro=1 drv=host_device
 floppy0: type=floppy removable=1 locked=0 [not inserted]
 sd0: type=floppy removable=1 locked=0 [not inserted]
 (qemu) change ide1-cd0 /path/to/my.iso
 (qemu) change ide1-cd0 /dev/sr0 host_device

===eject===
* <tt>eject [-f] ''device''</tt>
Use the <code>eject</code> command to release the device or file connected to the removable media device specified. The <code>-f</code> parameter can be used to force it if it initially refuses!

===usb_add===
Add a host file as USB flash device
( you need to create in advance the host file: dd if=/dev/zero of=/tmp/disk.usb bs=1024k count=32 )

usb_add disk:/tmp/disk.usb

===usb_del===
use info usb to get the usb device list
<br>
(qemu)info usb<br>
Device 0.1, Speed 480 Mb/s, Product XXXXXX<br>
Device 0.2, Speed 12 Mb/s, Product XXXXX<br>
<br>
(qemu)usb_del 0.2<br>
<br>
This deletes the device

===mouse_move===
Sends Mouse Movevment events to guest.
mouse_move dx dy [dz] -- send mouse move events.
Example:
[qemu]mouse_move -20 20

===mouse_button===

===mouse_set index===

===sendkey keys===
You can emulate keyboard events through sendkey command. The syntax is: sendkey keys. To get a list of keys, type <tt>sendkey [tab]</tt>. Examples: 
* <tt>sendkey a</tt>
* <tt>sendkey shift-a</tt>
* <tt>sendkey ctrl-u</tt>
* <tt>sendkey ctrl-alt-f1</tt>

As of QEMU 0.12.5 there are:

{| border=0
|shift
|shift_r
|alt
|alt_r
|altgr
|altgr_r
|-
|ctrl
|ctrl_r
|menu
|esc
|1
|2
|-
|3
|4
|5
|6
|7
|8
|-
|9
|0
|minus
|equal
|backspace
|tab
|-
|q
|w
|e
|r
|t
|y
|-
|u
|i
|o
|p
|ret
|a
|-
|s
|d
|f
|g
|h
|j
|-
|k
|l
|z
|x
|c
|v
|-
|b
|n
|m
|comma
|dot
|slash
|-
|asterisk
|spc
|caps_lock
|f1
|f2
|f3
|-
|f4
|f5
|f6
|f7
|f8
|f9
|-
|f10
|num_lock
|scroll_lock
|kp_divide
|kp_multiply
|kp_subtract
|-
|kp_add
|kp_enter
|kp_decimal
|sysrq
|kp_0
|kp_1
|-
|kp_2
|kp_3
|kp_4
|kp_5
|kp_6
|kp_7
|-
|kp_8
|kp_9
|<
|f11
|f12
|print
|-
|home
|pgup
|pgdn
|end
|left
|up
|-
|down
|right
|insert
|delete
|-
|}

==Screen and audio grabs==

===screendump===
* <tt>screendump ''filename''</tt>
Capture a screendump and save into a [[w:Netpbm format|PPM]] image file.

===wavcapture===

===stopcapture===

==Virtual machine==

===commit===
* <tt>commit ''device''</tt> or <tt>commit all</tt>
When running QEMU with the <code>-snapshot</code> option, commit changes to the device, or all devices.

===quit===
* <tt>quit</tt> or <tt>q</tt>
Quit QEMU '''immediately'''.

===savevm===
* <tt>savevm</tt> name
Save the virtual machine as the tag 'name'. Not all filesystems support this. raw does not, but qcow2 does.

===loadvm===
* <tt>loadvm</tt> name
Load the virtual machine tagged 'name'. This can also be done on the command line: <tt>-loadvm</tt> name

With the <tt>info snapshots</tt> command, you can request a list of available machines.

===delvm===

Remove the virtual machine tagged 'name'.

===stop===
Suspend execution of VM

===cont===
Reverse a previous stop command - resume execution of VM.

===system_reset===
This has an effect similar to the physical reset button on a PC. Warning: Filesystems may be left in an unclean state.

===system_powerdown===
This has an effect similar to the physical power button on a modern PC. The VM will get an ACPI shutdown request and usually shutdown cleanly.

===log===
* <tt>log ''option''</tt>

===logfile===
* <tt>logfile ''filename''</tt>
Write logs to specified file instead of the default path, <code>/tmp/qemu.log</code> .

===gdbserver===
Starts a remote debugger session for the GNU debugger (gdb).  To connect to it from the host machine, run the following commands:
 shell$ gdb qemuKernelFile
 (gdb) target remote localhost:1234

===x===
''x /format address''

Displays memory at the specified virtual address using the specified format.

Refer to the xp section for details on ''format'' and ''address''.

===xp===
''xp /format address''

Displays memory at the specified physical address using the specified format.

''format'': Used to specify the output format the displayed memory.  The format is broken down as ''/[count][data_format][size]
* count: number of item to display (base 10)
* data_format: 'x' for hex, 'd' for decimal, 'u' for unsigned decimal, 'o' for octal, 'c' for char and 'i' for (disassembled) processor instructions
* size: 'b' for 8 bits, 'h' for 16 bits, 'w' for 32 bits or 'g' for 64 bits. On x86 'h' and 'w' can select instruction disassembly code formats.

''address'':
* Direct address, for example: 0x20000
* Register, for example: $eip

Example - Display 3 instructions on an x86 processor starting at the current instruction:
<pre>
(qemu) xp /3i $eip
</pre>

Example - Display the last 20 words on the stack for an x86 processor:
<pre>
(qemu) xp /20wx $esp
</pre>

===print===
Print (or p), evaluates and prints the expression given to it.
The result will be printed in hexadecimal, but decimal can also be used in the expression.
If the result overflows it will wrap around.
To use a the value in a CPU register use $<register name>.  
The name of the register should be lower case.  You can see registers with the info registers command.

Example of qemu simulating an i386.
 (qemu) print 16
 0x10
 (qemu) print 16 + 0x10
 0x20
 (qemu) print $eax
 0xc02e4000
 (qemu) print $eax + 2
 0xc02e4000
 (qemu) print ($eax + 2) * 2
 0x805c8004
 (qemu) print 0x80000000 * 2
 0

More information on the architecture specific register names can be found from the below qemu source file

http://git.qemu.org/?p=qemu.git;a=blob;f=monitor.c;h=1266ba06fb032cb0e7c9dbaa1b6d22cd9047c6b4;hb=HEAD#l3044

===sum===

===memsave===
'''Usage:''' memsave <address> <size> <filename>

==Links==
Monitor in QEMU documentation: http://wiki.qemu.org/download/qemu-doc.html#pcsys_005fmonitor
{{Auto category}}
```

The resulting data:

```
{ type: 'page',
  text: 
   { Intro: 
      [ { text: 'When QEMU is running, it provides a monitor console for interacting with QEMU.',
          links: undefined },
        { text: 'Through various commands, the monitor allows you to inspect the running guest OS, change removable media and USB devices, take screenshots and audio grabs, and control various aspects of the virtual machine.',
          links: undefined },
        { text: 'The monitor is accessed from within QEMU by holding down the Ctrl and Alt keys (or whatever the "mouse grab" keystrokes are), and pressing  will send the monitor to the standard output, this is most useful when using qemu on the command line.',
          links: undefined } ],
     help: 
      [ { text: 'With no arguments, the help command lists all commands available.',
          links: undefined },
        { text: 'For more detail about another command, type help command, e.g.',
          links: undefined },
        { text: '(qemu) help info', links: undefined },
        { text: 'On a small screen / VM window, the list of commands will scroll off the screen too quickly to let you read them.',
          links: undefined },
        { text: 'To scroll back and forth so that you can read the whole list, hold down the control key and press Page Up and Page Down.',
          links: undefined } ],
     info: 
      [ { text: 'Show information on some aspect of the guest OS. Available options are:',
          links: undefined } ],
     change: 
      [ { text: 'The  to make the change.', links: undefined },
        { text: '(qemu) info block', links: undefined },
        { text: 'ide0-hd0: type=hd removable=0 file=/path/to/winxp.img',
          links: undefined },
        { text: 'ide0-hd1: type=hd removable=0 file=/path/to/pagefile.raw',
          links: undefined },
        { text: 'ide1-hd1: type=hd removable=0 file=/path/to/testing_data.img',
          links: undefined },
        { text: 'ide1-cd0: type=cdrom removable=1 locked=0 file=/dev/sr0 ro=1 drv=host_device',
          links: undefined },
        { text: 'floppy0: type=floppy removable=1 locked=0 [not inserted]',
          links: undefined },
        { text: 'sd0: type=floppy removable=1 locked=0 [not inserted]',
          links: undefined },
        { text: '(qemu) change ide1-cd0 /path/to/my.iso',
          links: undefined },
        { text: '(qemu) change ide1-cd0 /dev/sr0 host_device',
          links: undefined } ],
     eject: 
      [ { text: 'Use the  parameter can be used to force it if it initially refuses!',
          links: undefined } ],
     usb_add: 
      [ { text: 'Add a host file as USB flash device',
          links: undefined },
        { text: '( you need to create in advance the host file: dd if=/dev/zero of=/tmp/disk.usb bs=1024k count=32 )',
          links: undefined },
        { text: 'usb_add disk:/tmp/disk.usb', links: undefined } ],
     usb_del: 
      [ { text: 'use info usb to get the usb device list',
          links: undefined },
        { text: '(qemu)info usb', links: undefined },
        { text: 'Device 0.1, Speed 480 Mb/s, Product XXXXXX',
          links: undefined },
        { text: 'Device 0.2, Speed 12 Mb/s, Product XXXXX',
          links: undefined },
        { text: '(qemu)usb_del 0.2', links: undefined },
        { text: 'This deletes the device', links: undefined } ],
     mouse_move: 
      [ { text: 'Sends Mouse Movevment events to guest.',
          links: undefined },
        { text: 'mouse_move dx dy [dz] send mouse move events.',
          links: undefined },
        { text: 'Example:', links: undefined },
        { text: '[qemu]mouse_move -20 20', links: undefined } ],
     'sendkey keys': 
      [ { text: 'You can emulate keyboard events through sendkey command.',
          links: undefined },
        { text: 'The syntax is: sendkey keys.', links: undefined },
        { text: 'To get a list of keys, type sendkey [tab] .',
          links: undefined },
        { text: 'Examples:', links: undefined },
        { text: 'As of QEMU 0.12.5 there are:', links: undefined } ],
     screendump: 
      [ { text: 'Capture a screendump and save into a PPM image file.',
          links: [ { page: 'W:Netpbm format', src: 'PPM' } ] } ],
     commit: 
      [ { text: 'When running QEMU with the  option, commit changes to the device, or all devices.',
          links: undefined } ],
     quit: [ { text: 'Quit QEMU immediately.', links: undefined } ],
     savevm: 
      [ { text: 'Save the virtual machine as the tag \'name\'.',
          links: undefined },
        { text: 'Not all filesystems support this.', links: undefined },
        { text: 'raw does not, but qcow2 does.', links: undefined } ],
     loadvm: 
      [ { text: 'Load the virtual machine tagged \'name\'.',
          links: undefined },
        { text: 'This can also be done on the command line: -loadvm name',
          links: undefined },
        { text: 'With the info snapshots command, you can request a list of available machines.',
          links: undefined } ],
     delvm: 
      [ { text: 'Remove the virtual machine tagged \'name\'.',
          links: undefined } ],
     stop: [ { text: 'Suspend execution of VM', links: undefined } ],
     cont: 
      [ { text: 'Reverse a previous stop command - resume execution of VM.',
          links: undefined } ],
     system_reset: 
      [ { text: 'This has an effect similar to the physical reset button on a PC. Warning: Filesystems may be left in an unclean state.',
          links: undefined } ],
     system_powerdown: 
      [ { text: 'This has an effect similar to the physical power button on a modern PC. The VM will get an ACPI shutdown request and usually shutdown cleanly.',
          links: undefined } ],
     logfile: 
      [ { text: 'Write logs to specified file instead of the default path,  .',
          links: undefined } ],
     gdbserver: 
      [ { text: 'Starts a remote debugger session for the GNU debugger (gdb).',
          links: undefined },
        { text: 'To connect to it from the host machine, run the following commands:',
          links: undefined },
        { text: 'shell$ gdb qemuKernelFile', links: undefined },
        { text: '(gdb) target remote localhost:1234', links: undefined } ] },
  categories: [],
  images: [],
  infobox: {},
  infobox_template: '',
  tables: 
   [ [ [ 'ctrl', 'ctrl_r', 'menu', 'esc', '1', '2' ],
       [ '3', '4', '5', '6', '7', '8' ],
       [ '9', '0', 'minus', 'equal', 'backspace', 'tab' ],
       [ 'q', 'w', 'e', 'r', 't', 'y' ],
       [ 'u', 'i', 'o', 'p', 'ret', 'a' ],
       [ 's', 'd', 'f', 'g', 'h', 'j' ],
       [ 'k', 'l', 'z', 'x', 'c', 'v' ],
       [ 'b', 'n', 'm', 'comma', 'dot', 'slash' ],
       [ 'asterisk', 'spc', 'caps_lock', 'f1', 'f2', 'f3' ],
       [ 'f4', 'f5', 'f6', 'f7', 'f8', 'f9' ],
       [ 'f10',
         'num_lock',
         'scroll_lock',
         'kp_divide',
         'kp_multiply',
         'kp_subtract' ],
       [ 'kp_add', 'kp_enter', 'kp_decimal', 'sysrq', 'kp_0', 'kp_1' ],
       [ 'kp_2', 'kp_3', 'kp_4', 'kp_5', 'kp_6', 'kp_7' ],
       [ 'kp_8', 'kp_9', '<', 'f11', 'f12', 'print' ],
       [ 'home', 'pgup', 'pgdn', 'end', 'left', 'up' ],
       [ 'down', 'right', 'insert', 'delete' ],
       [],
       '-1': [ 'shift', 'shift_r', 'alt', 'alt_r', 'altgr', 'altgr_r' ] ] ],
  translations: {} }
```

**Comment by spencermountain** [comment]
ah, hi Sven. good find.
looks like it trips up here:
```
run the following commands:
 shell$ gdb qemuKernelFile
 (gdb) target remote localhost:1234
```
i've got a pretty busy couple weeks, and may not be able to get to this. 
Open to any PRs, folks!

**Comment by spencermountain** [comment]
hey, this is fixed now in v0.8.5
cheers
